app/controllers/concerns/consent_form_mailer_concern.rb

```diff
@@ -58,4 +58,39 @@ def send_consent_form_confirmation_refused(consent_form, programmes: nil)
       SMSDeliveryJob.perform_later(:consent_confirmation_refused, **params)
     end
   end
+
+  def send_parental_contact_warning_if_needed(patient, consent_form, user)
+    return if patient.parents.empty? || user.present?
```

Comment:
I think it's slightly confusing to have the check on the `user` being present here, especially as it's no then used in the method. I would suggest changing the way this method is called:

```rb
if current_user.nil?
  send_parental_contact_warning_if_needed(patient, self)
end
```